tests/test_annotated.py

```diff
@@ -0,0 +1,136 @@
+import sys
+from typing import get_type_hints
+
+import pytest
+
+from pydantic import BaseModel, Field
+from pydantic.fields import Undefined
+from pydantic.typing import Annotated
+
+
+@pytest.mark.skipif(not Annotated, reason='typing_extensions not installed')
+@pytest.mark.parametrize(
```

Comment:
I think this is okay, but as a piece of feedback, it's generally a good idea to include a simple test that demonstrates the basic behaviour at the top of test file. All this `parametrize` stuff is very clever but it's not that easy to read or reason with when things break.

You could also use [`pytest.importorskip`](https://docs.pytest.org/en/latest/skipping.html#skipping-on-a-missing-import-dependency) here to avoid lots of `skipif`, or just [`pytestmark = ...`](https://docs.pytest.org/en/latest/skipping.html#summary).